pkg/apis/istio/v1alpha2/validation/validation.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"fmt"
+	"reflect"
+
+	"istio.io/operator/pkg/apis/istio/v1alpha2"
+	"istio.io/operator/pkg/util"
+)
+
+const (
+	validationMethodName = "Validate"
+)
+
+// ValidateConfig  calls validation func for every defined element in Values
+func ValidateConfig(failOnMissingValidation bool, values *v1alpha2.Values, icpls *v1alpha2.IstioControlPlaneSpec) util.Errors {
+	var validationErrors util.Errors
+
+	validationErrors = util.AppendErrs(validationErrors, validateSubTypes(reflect.ValueOf(values).Elem(), failOnMissingValidation, values, icpls))
+
+	return validationErrors
+}
+
+func validateSubTypes(e reflect.Value, failOnMissingValidation bool, values *v1alpha2.Values, icpls *v1alpha2.IstioControlPlaneSpec) util.Errors {
+	var validationErrors util.Errors
+	var ptr reflect.Value
+	var value reflect.Value
+	var object reflect.Value
+	var finalMethod reflect.Value
+
+	// Dealing with receiver pointer and receiver value
+	if e.Type().Kind() == reflect.Ptr {
+		ptr = e
+		value = ptr.Elem()
+		object = reflect.Indirect(e)
+	} else {
+		ptr = reflect.New(reflect.TypeOf(e.Interface()))
+		temp := ptr.Elem()
+		temp.Set(e)
+		object = e
+		value = e
+	}
```

Comment:
Since Elem() handles both ptr and interface cases, you can just write:
k := e.Kind()
if k == reflect.Ptr || k == reflect.Interface {
  e = e.Elem()
}

Reply:
Sorry I do not follow this comment, I suspect it was meant for different line.

Reply:
let's chat about it in person - unfortunately next week is going to be a bit crazy because the code freeze on the 14th. I'll ping you soon after to go over this.

Reply:
still outstanding - you can either incorporate the change and we can merge or we can chat offline.